includes/User.php

```diff
@@ -4851,11 +4851,15 @@ protected function saveOptions() {
 		if ($wgPreferencesUseService) {
 			$preferenceNames = array_keys($this->userPreferences()->getPreferences($this->getId()));
 
-			(new WikiaSQL())
+			$sql = (new WikiaSQL())
 				->DELETE('user_properties')
-				->WHERE('up_user')->EQUAL_TO($this->getId())
-					->AND_('up_property')->NOT_IN($preferenceNames)
-				->run($dbw);
+				->WHERE('up_user')->EQUAL_TO($this->getId());
+
+			if (!empty($preferenceNames)) {
+				$sql->AND_('up_property')->NOT_IN($preferenceNames);
+			}
+
+			$sql->run($dbw);
```

Comment:
Making a DELETE query in `saveOptions` method is a bit counter-intuitive :) Can we add a comment here describing why we're doing that?